src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -213,6 +213,10 @@ private ExportResult OnCollect(Batch<Metric> metrics)
                     }
                 }
             }
+            else
+            {
+                this.targetInfoBufferLength = -1;
```

Comment:
@hez2010 Would you please add a note in the CHANGELOGs for Prometheus projects about this? Also, could we add a test?

Reply:
Done.